wifi: Migrate to nRF70 upstream (#16753)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst

This section describes the changes related to samples and applications.

> the nRF70 Series

Reviewer: I think we should clearly mention that customer should replace `CONFIG_WIFI_NRF700X*` Kconfig options with `CONFIG_WIFI_NRF700X*` Kconfig options. See below how clearly functions and Kconfig option is mentioned for lte_lc.

Author: Does it have to be an exhaustive list? They are too many :-(.

Author: discussed offline with @richabp we both felt it's better leave it to generic as there are too many Kconfigs and its a trivial namespace rename.

Author: Resolving this but if you disagree feel free to unresolve and we can discuss.

Reviewer: I wasn't looking for a big list but a version with `*` character which I wrote in my original comment.
I'm trying to make migrations easier as we see complaints that migrations from one release to another are painful but we still continue to cause more and more changes during the migrations. This is a trivial namespace change but if we can save 5mins for one unexperienced developer, it's worth it (and I honestly believe we save a lot more no matter how trivial this feels).
But up to you and tech writers.

Author: Agreed, I have elaborated with an example, is that fine? https://github.com/nrfconnect/sdk-nrf/pull/16753/files#diff-73508775c4364c7221f61e1b098cd77eaf956e37eb209b707c649bc2b9a987c7.

Reviewer: Looks good to me!

> the nRF70 Series
> ----------------
> 

Reviewer 2: ```suggestion
nRF70 Series
------------

```

.. toggle::

>    * The nRF70 Series support is now part of Zephyr upstream and it warrants below changes
> 
>     * The nRF70 Series driver namespace has been renamed from ``NRF700X`` to ``NRF70``.
>       Update your application to use the new namespace.
> 
>     * The nRF70 Series driver now uses per-module kernel heap with a higher default.
>       If a sample or an application uses the kernel heap but uses less than the default size, a build warning is displayed.
>       Use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` and enable it to suppress the warning.

Reviewer 2: ```suggestion
   * The nRF70 Series support is now part of Zephyr upstream, necessitating the following changes:

     * The nRF70 Series driver namespace has been renamed from ``NRF700X`` to ``NRF70``.
       Update your application to use the new namespace.
     * The nRF70 Series driver now uses a per-module kernel heap with a higher default.
      If a sample or an application uses the kernel heap less than the default size, a build warning is displayed.
      Use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` Kconfig option and enable it to suppress the warning.
```

    * ``AT#XSOCKETOPT=1,53,`` with ``AT#XSOCKETOPT=1,61,4`` to indicate ``RAI_ONGOING``.
    * ``AT#XSOCKETOPT=1,54,`` with ``AT#XSOCKETOPT=1,61,5`` to indicate ``RAI_WAIT_MORE``.

nRF70 Series
------------

.. toggle::

   * The nRF70 Series support is now part of Zephyr upstream and it warrants below changes:

    * The nRF70 Series driver namespace has been renamed from ``NRF700X`` to ``NRF70``.
      Update your application to use the new namespace.
    * The nRF70 Series driver now uses per-module kernel heap with a higher default.
    If a sample or an application uses the kernel heap but uses less than the default size, a build warning is displayed.

>     Use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` and enable it to suppress the warning.

Reviewer 2: ```suggestion
    Use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` Kconfig option and enable it to suppress the warning.
```
